tests/test_regr.py

```diff
@@ -139,3 +141,29 @@ def test_pylint_config_attr() -> None:
     assert len(inferred) == 1
     assert inferred[0].root().name == "optparse"
     assert inferred[0].name == "Values"
+
+
+@contextmanager
+def timeout(timeout_s):
+    def _handle(_signum, _frame):
+        pytest.fail("timed out")
+
+    signal.signal(signal.SIGALRM, _handle)
+    signal.setitimer(signal.ITIMER_REAL, timeout_s)
+    yield
+    signal.setitimer(signal.ITIMER_REAL, 0)
+    signal.signal(signal.SIGALRM, signal.SIG_DFL)
+
+
+@pytest.mark.slow
+@pytest.mark.serial
+@pytest.mark.skipif(not hasattr(signal, "setitimer"), reason="Assumes POSIX signals")
+@pytest.mark.parametrize(
+    "fname,timeout_s",
+    [
+        (join(REGR_DATA, "hang", "pkg4972.flask"), 30.0),
+    ],
+)
+def test_hang(finalize_linter, fname, timeout_s):
```

Comment:
My knowledge of `purest` is limited, but looking at the code and parameter names I guess the `30.0` means this test should pass in 30 seconds? If so, is that reasonable for our CI? Do you know how long this test normally takes?

Reply:
Yes I think 30s is a long time to add to the total time of every pipeline. If we can't do anything else (because it's hard to reproduce the problem), we might have to parallelize the tests marked slow in github actions. But I guess we could also make the performance of pylint better in the first place ?

Reply:
I think it fails whenever the test takes more than 30 seconds? Instead of adding 30 seconds every time.
However, I wonder if there are moments at which GitHub lags too much and the test takes more than 30 seconds and we fail as false positive.

Reply:
The test fails if it takes more than 30s yes, it only takes a couple of seconds on my 2014-era laptop so I just picked a 10x timeout for safety. Parallelising them wouldn't be compatible with the SIGALRM approach so that would have to change if you start adding `pytest-xdist` or something in the future.